collector/src/bin/rustc-perf-collector/execute.rs

```diff
@@ -276,7 +281,11 @@ impl<'a> CargoProcess<'a> {
                     ));
                 }
 
-                profiler.subcommand()
+                if self.build_kind == BuildKind::Doc {
+                    "rustdoc"
+                } else {
+                    profiler.subcommand()
+                }
```

Comment:
This feels a bit not nice. Can we instead pass in the build kind, and match on that within the subcommand impls? We'd assert that the kind is debug/opt for the llvm-lines subcommand, I guess.

Reply:
>  We'd assert that the kind is debug/opt for the llvm-lines subcommand, I guess.

We already pass in the build kind for `is_build_kind_allowed`. Maybe we could combine that and `subcommand()`, so that subcommand now returns an Option? That would avoid asserts.

Reply:
Sure, that works too.